tests/functional/test_api_versions.py

```diff
@@ -49,3 +49,8 @@ def test_command_interface_reflects_api_version(self):
         cmdline = 'ec2 describe-nat-gateways'
         _, stderr, _ = self.run_cmd(cmdline, expected_rc=2)
         self.assertIn("Invalid choice: 'describe-nat-gateways'", stderr)
+
+    def test_handles_ProfileNotFound_exception(self):
```

Comment:
I'm thinking we do the following with this test:
1. We rename this to `test_can_get_version_with_non_existent_profile` to better capture the issues we are trying to prevent.
2. We rename the `test_args_resolution.py` to `test_args.py` and move that test there. In v2, the api version functionality all goes away so we need to find a home for this test. Also with the test name change, the test case is something a bit more general then api versions.